src/amber/router/parsers/multipart.cr

```diff
@@ -1,16 +1,32 @@
 module Amber::Router::Parsers
   module Multipart
-    def self.parse(request : HTTP::Request) : Tuple(Types::Params, Types::Files)
+    def self.parse(request : HTTP::Request) : Tuple(Types::Params, Types::File | Types::Files)
       multipart_params = Types::Params.new
-      files = Types::Files.new
+      files = Types::File.new
 
       HTTP::FormData.parse(request) do |upload|
         next unless upload
+
         filename = upload.filename
         if filename.is_a?(String) && !filename.empty?
-          files[upload.name] = Amber::Router::File.new(upload: upload)
+          if files.has_key? upload.name
```

Comment:
I'm struggling to follow this logic.  Can you refactor this down into smaller methods?

Reply:
I agree we can separate these?

Reply:
I'll trying to refactor it. Thank you.